src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx

```diff
@@ -297,6 +298,36 @@ export const SubstrateHomeAdaptorProvider = ({
     [api, setDepositNonce, setTransactionStatus, address, homeChainConfig]
   );
 
+  const handleCheckSupplies = useCallback(
+    async (
+      amount: number,
+      tokenAddress: string,
+      destinationChainId: number
+    ) => {
+      if (homeChainConfig) {
+        const destinationChain = getСhainConfig(destinationChainId);
+        const token = homeChainConfig.tokens.find(
+          (token) => token.address === tokenAddress
+        );
+
+        if (destinationChain?.type === "Ethereum" && token) {
+          const hasSupplies = await hasTokenSupplies(
+            destinationChain,
+            tokens,
+            token,
+            amount,
+            tokenAddress
+          );
+          if (!hasSupplies) {
+            return false;
+          }
+        }
+        return true;
```

Comment:
if `destinationChain?.type` is not `Ethereum` we should log error and return `false`.